File: 1.Listas_enlazadas/C/libro.c
//
// libro.c
// EstructurasC
//
// Creado por José Carlos Rodríguez Cañas
// Copyright (c) 2026 José Carlos Rodríguez Cañas
//

#include <stdio.h>
#include <string.h>
#include "libro.h"

// Función auxiliar para leer un libro desde stdin
void leerLibro(Libro *lib) {
	printf("Ingrese el titulo del libro: ");
	if (!fgets(lib->titulo, sizeof lib->titulo, stdin)) lib->titulo[0] = '\0';
	lib->titulo[strcspn(lib->titulo, "\n")] = '\0';

	printf("Ingrese el autor del libro: ");
	if (!fgets(lib->autor, sizeof lib->autor, stdin)) lib->autor[0] = '\0';
	lib->autor[strcspn(lib->autor, "\n")] = '\0';

	printf("Ingrese el ISBN del libro: ");
	if (!fgets(lib->isbn, sizeof lib->isbn, stdin)) lib->isbn[0] = '\0';
	lib->isbn[strcspn(lib->isbn, "\n")] = '\0';
}

File: 1.Listas_enlazadas/Java/Libro.java
//
// Libro.java
// EstructurasJava
//
// Creado por José Carlos Rodríguez Cañas
// Copyright (c) 2026 José Carlos Rodríguez Cañas
//

import java.util.Scanner;

// Clase para representar un libro con título, autor e ISBN 
public class Libro {

    private String titulo;
    private String autor;
    private String isbn;

    // Constructor por defecto
    public Libro() {
        this.titulo = "";
        this.autor = "";
        this.isbn = "";
    }

    // Constructor con parámetros
    public Libro(String titulo, String autor, String isbn) {
        this.titulo = titulo;
        this.autor = autor;
        this.isbn = isbn;
    }

    // Métodos Getter y Setter
    public String getTitulo() {
        return titulo;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    // Función auxiliar para leer un libro desde stdin
    public static void leerLibro(Libro lib, Scanner scanner) {
        System.out.print("Ingrese el titulo del libro: ");
        lib.setTitulo(scanner.nextLine());

        System.out.print("Ingrese el autor del libro: ");
        lib.setAutor(scanner.nextLine());

        System.out.print("Ingrese el ISBN del libro: ");
        lib.setIsbn(scanner.nextLine());
    }
}